### ISSUE TITLE
Cannot access member "specifier" for type "Requirement" 

### ISSUE THREAD

**Issue by joao-parana** [microsoft/pyright#5377]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**

When I open my python file in VSCode or when I run upright CLI in Terminal I get the error:

`Cannot access member "specifier" for type "Requirement" `.

See line 18 in my python code bellow !

**To Reproduce**
Steps to reproduce the behavior.

1. put the python code in python file2. 2. 
2. open the python file in VSCode or run pyright CLI
3. see the error: `error: Cannot access member "specifier" for type "Requirement".  Member "specifier" is unknown (reportGeneralTypeIssues)`

**Expected behavior**

`No errors should be displayed because "specifier" is a member of type "Requirement"`  of  `pkg_resources` !

**Code or Screenshots**
If possible, provide a minimal, self-contained code sample (surrounded by triple back ticks) to demonstrate the issue. The code should define or imports all referenced symbols.

```python
import pkg_resources     # Python 3.11.3
from pkg_resources import Requirement
. . . 
def check_missing_packages() -> bool:
    requirements_file = 'requirements.txt' # sys.argv[1]
    with open(requirements_file, "r") as f:
        required_packages = [
            line.strip().split("#")[0].strip() for line in f.readlines()
        ]
    working_set = pkg_resources.working_set
    installed_packages = {pkg.key: pkg.version for pkg in working_set}
    missing_packages = []
    print('-------------------- Requirements --------------------')
    for required_package in required_packages:
        if not required_package:  # Skip empty lines
            continue
        pkg: Requirement = Requirement.parse(required_package)
        pkg_specifier = pkg.specifier                           # ERROR REPORTED IN THIS LINE
        suffix_message = f', version: {pkg_specifier}' if len(str(pkg_specifier)) > 1 else ''
        print(f'pkg = {pkg.key}' + suffix_message)
        if (pkg.key not in installed_packages
            or pkg_resources.parse_version(
                    installed_packages[pkg.key]) not in pkg_specifier):
            missing_packages.append(str(pkg))
    if missing_packages:
        print("Missing packages Error: Aborting")
        print(", ".join(missing_packages))
        return False
    else:
        print("All packages are installed.")
        return True
```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension, a language server in another editor, or the command-line tool? Which version?

VSCode extension: Pyright - v1.1.315
pyright CLI : pyright - 1.1.315

**Additional context**

**Comment by erictraut** [comment]
The type information for the `setuptools` library comes from the typeshed project, which is a community-maintained collection of type stubs for the Python stdlib and various third-party libraries. Pyright ships with a bundled version of the typeshed stubs. 

The `Requirement` type that you're accessing in your code is defined [here](https://github.com/python/typeshed/blob/575dd0c448fed4e650b4a863a4d9160f6e092267/stubs/setuptools/pkg_resources/__init__.pyi#L86). As you can see, there's no attribute called `specifier`. There is an attribute called `specs`. Is that what you're looking for? If an attribute called `specifier` is present at runtime, then it's likely just an omission in the type stub. You could file a bug (or even a PR with a fix) in the typeshed project.

I'm going to close this because pyright is doing the right thing here given the type information it is provided.
